lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -262,25 +266,33 @@ def should_cleanup?
 
     def session(command)
       @session = true
-      command_out, slave = PTY.open
-      command_in, write = IO.pipe
-      command_pid = spawn(({'SSHPASS' => @ssh_password} if @ssh_password), '/usr/bin/sshpass', '-e', '/usr/bin/ssh', @host, *ssh_options, command, :in => command_in, :out => slave)
-      return command_in, command_out, command_pid, slave, write
+
+      in_read, in_write = IO.pipe
+      out_read, out_write = IO.pipe
+      command_pid = if @ssh_password 
+                      spawn({'SSHPASS' => @ssh_password}, '/usr/bin/sshpass', '-e', '/usr/bin/ssh', @host, *ssh_options, command, :in => in_read, :out => out_write)
+                    else
+                      spawn('/usr/bin/ssh', @host, *ssh_options, command, :in => in_read, :out => out_write)
+                    end
```

Comment:
We can leverage splatting to reduce repetition


```suggestion
      args = []
      args += [{'SSHPASS' => @ssh_password}, '/usr/bin/sshpass', '-e']) if @ssh_password
      args += ['/usr/bin/ssh', @host, ssh_options, command].flatten
      command_pid = spawn(*args, :in => in_read, :out => out_write)
```